app/Platform/Actions/UpsertGameCoreAchievementSetFromLegacyFlagsAction.php

```diff
@@ -119,6 +119,9 @@ private function updateExistingCoreSetOfGame(Game $game): void
     private function syncAchievementSetAchievements(AchievementSet $coreSet, Collection $syncData): void
     {
         if ($syncData->isEmpty()) {
+            // Delete achievements that are no longer in the set.
+            AchievementSetAchievement::where('achievement_set_id', $coreSet->id)->delete();
+
```

Comment:
Mimics behavior from a few lines down when a non-zero number of achievements are still present.